labelbox/client.py

```diff
@@ -184,20 +186,15 @@ def convert_value(value):
         endpoint = self.endpoint if not experimental else self.endpoint.replace(
             "/graphql", "/_gql")
 
-        try:
-            request = {
-                'url': endpoint,
-                'data': data,
-                'headers': self.headers,
-                'timeout': timeout
-            }
-            if files:
-                request.update({'files': files})
-                request['headers'] = {
-                    'Authorization': self.headers['Authorization']
-                }
+        request = requests.Request(method="POST", url=endpoint, data=data)
+        prepare_req = request.prepare()
```

Comment:
what is the advantage of using a prepared request here as compared to how it was done before?

Reply:
Also I have found this comment [here](https://requests.readthedocs.io/en/latest/user/advanced/#prepared-requests):
>However, the above code will lose some of the advantages of having a Requests [Session](https://requests.readthedocs.io/en/latest/api/#requests.Session) object. In particular, [Session](https://requests.readthedocs.io/en/latest/api/#requests.Session)-level state such as cookies will not get applied to your request. To get a [PreparedRequest](https://requests.readthedocs.io/en/latest/api/#requests.PreparedRequest) with that state applied, replace the call to [Request.prepare()](https://requests.readthedocs.io/en/latest/api/#requests.Request.prepare) with a call to [Session.prepare_request()](https://requests.readthedocs.io/en/latest/api/#requests.Session.prepare_request), like this: